resources/providers/azurelib/inventory/provider.go

```diff
@@ -224,3 +259,75 @@ func getString(data map[string]any, key string) string {
 	value, _ := data[key].(string)
 	return value
 }
+
+func transformDiagnosticSettingsClientListResponses(res []armmonitor.DiagnosticSettingsClientListResponse, subID, subName string) ([]AzureAsset, error) {
+	var assets []AzureAsset
+
+	for i := range res {
+		for _, v := range res[i].Value {
+			if v == nil {
+				continue
+			}
+			a, err := transformDiagnosticSettingsResource(v, subID, subName)
+			if err != nil {
+				return nil, fmt.Errorf("error parsing azure asset model: %w", err)
+			}
+			assets = append(assets, a)
+		}
+	}
+
+	return assets, nil
+}
+
+func transformDiagnosticSettingsResource(v *armmonitor.DiagnosticSettingsResource, subID, subName string) (AzureAsset, error) {
+	properties, err := transformDiagnosticSettings(v.Properties)
+	if err != nil {
+		return AzureAsset{}, err
+	}
+
+	return AzureAsset{
+		Id:               strings.Dereference(v.ID),
+		Name:             strings.Dereference(v.Name),
+		Location:         "",
+		Properties:       properties,
+		ResourceGroup:    "",
+		SubscriptionId:   subID,
+		SubscriptionName: subName,
+		TenantId:         "",
+		Sku:              "",
+		Type:             strings.Dereference(v.Type),
+	}, nil
+}
+
+func transformDiagnosticSettings(d *armmonitor.DiagnosticSettings) (map[string]any, error) {
+	if d == nil {
+		return nil, nil
+	}
+
+	js, err := d.MarshalJSON()
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal json: %w", err)
+	}
+
+	m := map[string]any{}
+	err = json.Unmarshal(js, &m)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal json: %w", err)
+	}
+
+	return m, nil
+}
+
+func readPager[T any](ctx context.Context, pager *runtime.Pager[T]) ([]T, error) {
```

Comment:
I am not sure what's the value of flattening the pager? Instead of `for pager.More()` we have a `for _, value := res`, so same level of indentation but without the benefits of a pager + slightly more code.

Reply:
I am not sure I understood; this is a helper function to avoid performing this loop inside the loop of subscriptions. 

I feel by doing that the subscription loop is more readable.